lib/bundler/compatibility_guard.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: false
+
+require "rubygems"
+require "bundler/version"
+
+if Bundler::VERSION.split(".").first.to_i >= 2
+  if Gem::Version.new(Object::RUBY_VERSION.dup) < Gem::Version.new("2.3")
+    abort "Bundler 2 requires Ruby 2.3 or later. Either install bundler 1 or update to a supported Ruby version."
```

Comment:
I think it would be good to have a note on Bundler 1.x to suggest it's on a limited maintenance path to encourage the investment to upgrade.